runtime/win32.c

```diff
@@ -1086,20 +1086,20 @@ int caml_num_rows_fd(int fd)
 CAMLexport clock_t caml_win32_clock(void)
 {
   FILETIME _creation, _exit;
-  CAML_ULONGLONG_FILETIME stime, utime;
-  ULONGLONG clocks_per_sec;
+  CAML_ULONGLONG_FILETIME kernel_time, user_time;
+  uint64_t cpu_time_100_nsec;
 
   if (!(GetProcessTimes(GetCurrentProcess(), &_creation, &_exit,
-                        &stime.ft, &utime.ft))) {
+                        &kernel_time.ft, &user_time.ft))) {
     return (clock_t)(-1);
   }
 
   /* total in 100-nanosecond intervals (1e7 / CLOCKS_PER_SEC) */
-  clocks_per_sec = 10000000ULL / (ULONGLONG)CLOCKS_PER_SEC;
-  return (clock_t)((stime.ul + utime.ul) / clocks_per_sec);
+  cpu_time_100_nsec = kernel_time.ul + user_time.ul;
+  return (clock_t) (cpu_time_100_nsec / (NSEC_PER_SEC / 100 / CLOCKS_PER_SEC));
```

Comment:
I'm willing to believe that the new code is equivalent to the previous code, but I find both somewhat obscure. My intuition would be to write this as:

```c
cpu_time_nsec = (kernel_time.ul + urser_time.ul) * 100;
return (clock_t) (CLOCKS_PER_SEC * cpu_time_nsec / NSEC_PER_SEC);
```

Reply:
I think I'll withdraw this part. I've just found out that `clock_t` is a `long` on Windows, and `clock` is documented to return `-1` when the process has run for more that 24.8 days. I think we should change `caml_win32_clock` to return an `uint64_t` and change how we use it in the runtime. I looked at `caml_sys_time_include_children_unboxed`, and it doesn't handle errors, so callers of `Sys.times` get undefined or weird, overflowed, or negative values. I'll think about it and maybe I'll submit another PR.